diff --git a/src/lib/fcitx/addonmanager.cpp b/src/lib/fcitx/addonmanager.cpp
index 55269f33..5e7b8d27 100644
--- a/src/lib/fcitx/addonmanager.cpp
+++ b/src/lib/fcitx/addonmanager.cpp
@@ -250,7 +250,6 @@ void AddonManager::unregisterLoader(const std::string &name) {
 }
 
 void AddonManager::registerDefaultLoader(StaticAddonRegistry *registry) {
-    registerLoader(std::make_unique<SharedLibraryLoader>());
     if (registry) {
         registerLoader(std::make_unique<StaticLibraryLoader>(registry));
     }
