diff --git a/src/lib/fcitx/addonmanager.cpp b/src/lib/fcitx/addonmanager.cpp
index ab4e36f4..48001b4e 100644
--- a/src/lib/fcitx/addonmanager.cpp
+++ b/src/lib/fcitx/addonmanager.cpp
@@ -249,7 +249,6 @@ void AddonManager::unregisterLoader(const std::string &name) {
 }
 
 void AddonManager::registerDefaultLoader(StaticAddonRegistry *registry) {
-    registerLoader(std::make_unique<SharedLibraryLoader>());
     if (registry) {
         registerLoader(std::make_unique<StaticLibraryLoader>(registry));
     }
