diff --git a/CMakeLists.txt b/CMakeLists.txt
index bb3c56c..ea60442 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -230,7 +230,10 @@ else()
     set(FCITX_LIBRARY_SUFFIX ".so")
 endif()
 
-check_function_exists(pipe2 HAVE_PIPE2)
+set(CMAKE_REQUIRED_DEFINITIONS -D_GNU_SOURCE)
+check_symbol_exists(pipe2 "unistd.h" HAVE_PIPE2)
+unset(CMAKE_REQUIRED_DEFINITIONS)
+
 check_include_file(sys/wait.h HAVE_SYS_WAIT_H)
 check_include_file(sys/mman.h HAVE_SYS_MMAN_H)
 check_include_file(sys/uio.h HAVE_SYS_UIO_H)
diff --git a/src/lib/fcitx-utils/log.cpp b/src/lib/fcitx-utils/log.cpp
index 56fb7f6..98fa2ae 100644
--- a/src/lib/fcitx-utils/log.cpp
+++ b/src/lib/fcitx-utils/log.cpp
@@ -35,16 +35,20 @@ using LogRule = std::pair<std::string, LogLevel>;
 
 struct LogConfig {
     static std::ostream *defaultLogStream;
+#if 0
     static thread_local std::osyncstream localLogStream;
+#endif
     static bool showTimeDate;
 };
 
 std::ostream *LogConfig::defaultLogStream = &std::cerr;
+#if 0
 thread_local std::osyncstream LogConfig::localLogStream = []() {
     std::osyncstream out(*LogConfig::defaultLogStream);
     out.rdbuf()->set_emit_on_sync(true);
     return out;
 }();
+#endif
 bool LogConfig::showTimeDate = true;
 
 bool validateLogLevel(std::underlying_type_t<LogLevel> l) {
@@ -199,12 +203,15 @@ void Log::setLogStream(std::ostream &stream) {
 }
 
 std::ostream &Log::logStream() {
+#if 0
     auto *buf = LogConfig::defaultLogStream->rdbuf();
     if (LogConfig::localLogStream.get_wrapped() != buf) {
         LogConfig::localLogStream = std::osyncstream(buf);
         LogConfig::localLogStream.rdbuf()->set_emit_on_sync(true);
     }
     return LogConfig::localLogStream;
+#endif
+    return *LogConfig::defaultLogStream;
 }
 
 LogMessageBuilder::LogMessageBuilder(std::ostream &out, LogLevel l,
diff --git a/src/lib/fcitx/addonmanager.cpp b/src/lib/fcitx/addonmanager.cpp
index ab4e36f..48001b4 100644
--- a/src/lib/fcitx/addonmanager.cpp
+++ b/src/lib/fcitx/addonmanager.cpp
@@ -249,7 +249,6 @@ void AddonManager::unregisterLoader(const std::string &name) {
 }
 
 void AddonManager::registerDefaultLoader(StaticAddonRegistry *registry) {
-    registerLoader(std::make_unique<SharedLibraryLoader>());
     if (registry) {
         registerLoader(std::make_unique<StaticLibraryLoader>(registry));
     }
