diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index bf6da01..e117fa5 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,4 +1,4 @@
-add_library(hallelujah MODULE hallelujah.cpp factory.cpp)
+add_library(hallelujah STATIC hallelujah.cpp factory.cpp)
 target_link_libraries(hallelujah Fcitx5::Core Fcitx5::Module::Spell fmt::fmt ${MARISA_TARGET} ${JSONC_TARGET})
 install(TARGETS hallelujah DESTINATION "${FCITX_INSTALL_LIBDIR}/fcitx5")
 fcitx5_translate_desktop_file(hallelujah.conf.in hallelujah.conf)
