diff --git a/src/libime/core/CMakeLists.txt b/src/libime/core/CMakeLists.txt
index 35ee954..b1ada6a 100644
--- a/src/libime/core/CMakeLists.txt
+++ b/src/libime/core/CMakeLists.txt
@@ -69,7 +69,7 @@ ecm_setup_version(PROJECT
 # the cmake if will
 set(IMECore_SOVERSION 0)
 
-add_library(IMECore SHARED ${LIBIME_SRCS})
+add_library(IMECore STATIC ${LIBIME_SRCS})
 set_target_properties(IMECore
                       PROPERTIES VERSION ${IMECore_VERSION}
                       SOVERSION ${IMECore_SOVERSION}
@@ -82,7 +82,7 @@ target_include_directories(IMECore PUBLIC
 
 target_link_libraries(IMECore PUBLIC Fcitx5::Utils Boost::boost PRIVATE kenlm Boost::iostreams PkgConfig::ZSTD)
 
-install(TARGETS IMECore EXPORT LibIMECoreTargets LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT lib)
+install(TARGETS IMECore kenlm EXPORT LibIMECoreTargets LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT lib)
 install(FILES ${LIBIME_HDRS} DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/LibIME/libime/core" COMPONENT header)
 
 add_library(LibIME::Core ALIAS IMECore)
diff --git a/src/libime/pinyin/CMakeLists.txt b/src/libime/pinyin/CMakeLists.txt
index a5e6d35..601d559 100644
--- a/src/libime/pinyin/CMakeLists.txt
+++ b/src/libime/pinyin/CMakeLists.txt
@@ -32,7 +32,7 @@ ecm_setup_version(PROJECT
                   PACKAGE_VERSION_FILE "${CMAKE_CURRENT_BINARY_DIR}/LibIMEPinyinConfigVersion.cmake")
 set(IMEPinyin_SOVERSION 0)
 
-add_library(IMEPinyin SHARED ${LIBIME_PINYIN_SRCS})
+add_library(IMEPinyin STATIC ${LIBIME_PINYIN_SRCS})
 set_target_properties(IMEPinyin PROPERTIES
                       VERSION ${IMEPinyin_VERSION}
                       SOVERSION ${IMEPinyin_SOVERSION}
diff --git a/src/libime/table/CMakeLists.txt b/src/libime/table/CMakeLists.txt
index 62317bd..dd47d0b 100644
--- a/src/libime/table/CMakeLists.txt
+++ b/src/libime/table/CMakeLists.txt
@@ -23,7 +23,7 @@ ecm_setup_version(PROJECT
                   PACKAGE_VERSION_FILE "${CMAKE_CURRENT_BINARY_DIR}/LibIMETableConfigVersion.cmake")
 set(IMETable_SOVERSION 0)
 
-add_library(IMETable SHARED ${LIBIME_TABLE_SRCS})
+add_library(IMETable STATIC ${LIBIME_TABLE_SRCS})
 set_target_properties(IMETable
                       PROPERTIES VERSION ${IMETable_VERSION}
                       SOVERSION ${IMETable_SOVERSION}
