diff --git a/src/rimeengine.cpp b/src/rimeengine.cpp
index b335069..a65a123 100644
--- a/src/rimeengine.cpp
+++ b/src/rimeengine.cpp
@@ -47,6 +47,8 @@
 #include <utility>
 #include <vector>
 
+#include "fcitxipc-swift.h"
+
 FCITX_DEFINE_LOG_CATEGORY(rime_log, "rime");
 
 namespace fcitx::rime {
@@ -197,11 +199,17 @@ RimeEngine::RimeEngine(Instance *instance)
     deployAction_.setIcon("fcitx_rime_deploy");
     deployAction_.setShortText(_("Deploy"));
     deployAction_.connect<SimpleAction::Activated>([this](InputContext *ic) {
+#if 0
         deploy();
         auto *state = this->state(ic);
         if (state && ic->hasFocus()) {
             state->updateUI(ic, false);
         }
+#endif
+        std::string f5iEnv = getenv("F5I_ENV");
+        if (f5iEnv == "keyboard") {
+            FcitxIpc::openURL("fcitx://config/addon/rime/deploy");
+        }
     });
     instance_->userInterfaceManager().registerAction("fcitx-rime-deploy",
                                                      &deployAction_);
@@ -210,11 +218,17 @@ RimeEngine::RimeEngine(Instance *instance)
     syncAction_.setShortText(_("Synchronize"));
 
     syncAction_.connect<SimpleAction::Activated>([this](InputContext *ic) {
+#if 0
         sync();
         auto *state = this->state(ic);
         if (state && ic->hasFocus()) {
             state->updateUI(ic, false);
         }
+#endif
+        std::string f5iEnv = getenv("F5I_ENV");
+        if (f5iEnv == "keyboard") {
+            FcitxIpc::openURL("fcitx://config/addon/rime/sync");
+        }
     });
     instance_->userInterfaceManager().registerAction("fcitx-rime-sync",
                                                      &syncAction_);
@@ -305,7 +319,10 @@ void RimeEngine::rimeStart(bool fullcheck) {
     }
     api_->initialize(&fcitx_rime_traits);
     api_->set_notification_handler(&rimeNotificationHandler, this);
+    // Don't deploy on start in keyboard.
+    if (fullcheck) {
     api_->start_maintenance(fullcheck);
+    }
 
     if (!api_->is_maintenance_mode()) {
         updateAppOptions();
@@ -582,12 +599,14 @@ void RimeEngine::notify(RimeSessionId session, const std::string &messageType,
     bool blockMessage = false;
     if (messageType == "deploy") {
         tipId = "fcitx-rime-deploy";
-        icon = "fcitx_rime_deploy";
         if (messageValue == "start") {
             message = _("Rime is under maintenance. It may take a few "
                         "seconds. Please wait until it is finished...");
+            icon = "running";
         } else if (messageValue == "success") {
             message = _("Rime is ready.");
+            icon = "success";
+            FcitxIpc::requestReload();
             if (!api_->is_maintenance_mode()) {
                 if (needRefreshAppOption_) {
                     api_->deploy_config_file("fcitx5.yaml", "config_version");
@@ -602,6 +621,7 @@ void RimeEngine::notify(RimeSessionId session, const std::string &messageType,
             needRefreshAppOption_ = false;
             message = _("Rime has encountered an error. "
                         "See log for details.");
+            icon = "error";
             blockMessage = true;
         }
     } else if (messageType == "option") {
